src/radical/pilot/session.py

```diff
@@ -968,3 +985,29 @@ def excuse():
 
 
 # ------------------------------------------------------------------------------
+
+
+@dataclasses.dataclass(frozen=True)
+class _CloseOptions:
+    """Options and validation for Session.close().
+
+    **Arguments:**
+        * **cleanup**   (`bool`):
+          Remove session from MongoDB (implies * terminate). (default False)
+        * **terminate** (`bool`):
+          Shut down all pilots associated with the session. (default True)
+        * **download** (`bool`):
+          Fetch pilot profiles and database entries. (default False)
+
+    """
+    cleanup: bool = False
+    # Warning: Previously, passing `cleanup=None` would reverse the default and
+    # result in cleanup == True.
+
+    terminate: bool = True
+    download: bool = False
+
+    def __post_init__(self):
+        if self.cleanup and not self.terminate:
+            # cleanup implies terminate
+            raise ValueError('cleanup implies terminate.')
```

Comment:
There was previously no enforcement, so I may have misinterpreted the intention. Previously, if a user provided `cleanup=True` and `terminate=False`, this was not an error as far as the `close()` function call was concerned. I assume that was an oversight. If it was intentional, then this commit is wrong.